common/src/main/scala/com/gu/media/Permissions.scala

```diff
@@ -41,3 +51,11 @@ class MediaAtomMakerPermissionsProvider(stage: String, credsProvider: AWSCredent
     }
   }
 }
+
+object PermissionsUploadHelper {
+  def permissionAndIntentMatch(permission: Permissions, selfHosted: Boolean) : Boolean = {
```

Comment:
@mbarton yes I suspect we could have two routes but I wasn't sure of UI logic. Worth re-visiting when I do the work on showing the progress?